File: 74 (Search a 2D Matrix)/java/Solution.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int rows = matrix.length;
        int cols = matrix[0].length;

        int low = 0;
        int high = rows*cols-1;

        while (low<=high){
            int mid = (low+high)>>1;
            
            int cr = mid/cols;
            int cc = mid%cols;

            if (matrix[cr][cc] == target) return true;
            if (matrix[cr][cc] > target) high = mid-1;
            else low = mid +1;
        }
        return false;
    }
}

File: 74 (Search a 2D Matrix)/python/Solution.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        rows = len(matrix)
        cols = len(matrix[0])

        low = 0
        high = rows*cols-1

        while low<=high:
            mid= (low+high)>>1

            cr = mid//cols
            cc = mid%cols

            if matrix[cr][cc] == target: return True

            if matrix[cr][cc] > target: high =  mid-1
            else: low = mid+1
        
        return False